dm/syncer/dml_worker.go

```diff
@@ -117,24 +115,43 @@ func (w *DMLWorker) run() {
 
 	for j := range w.inCh {
 		metrics.QueueSizeGauge.WithLabelValues(w.task, "dml_worker_input", w.source).Set(float64(len(w.inCh)))
-		if j.tp == flush || j.tp == conflict {
-			if j.tp == conflict {
-				w.addCountFunc(false, adminQueueName, j.tp, 1, j.targetTable)
+		switch j.tp {
+		case flush:
```

Comment:
how about merge aysnc/syncflush and conflict like this ? it seems that most of the codes are same for both of them

```golang
		case flush, asyncFlush,conflict:
			w.addCountFunc(false, adminQueueName, j.tp, 1, j.targetTable)
			// flush for every DML queue
			for i, jobCh := range jobChs {
				startTime := time.Now()
				jobCh <- j
				metrics.AddJobDurationHistogram.WithLabelValues(j.tp.String(), w.task, queueBucketMapping[i], w.source).Observe(time.Since(startTime).Seconds())
			}
			if j.tp == asyncFlush  || j.tp == conflict{
				j.wg.Wait()
			}
			w.addCountFunc(true, adminQueueName, j.tp, 1, j.targetTable)
			w.flushCh <- j
```

Reply:
instead of merging it together, i have separate out the flush dml queue logic out as separate function to call.

the issue of merging all together is causing us to hard build the mental model about each case execution and hence raise the probabilities to introduce bugs. As presented on above code, it's easily to miss the point that we don't need to do `w.flushCh <- j` for conflict type job.

Reply:
btw, one more question here, how does this `w.addCountFunc(false, adminQueueName, j.tp, 1, j.targetTable)` work here? do we use it correctly here?

Reply:
>  it's easily to miss  the point that we don't need to do w.flushCh <- j for conflict type job.

sorry, mybad 😬

`addCountFunc` is used to update metrics, i think the current usage is mostly correct , but there is one point may  bother the user `j.targetTable` will always be empty string when jobType in [flush/asyncflush/confilct]